src/login/loginctl.c

```diff
@@ -232,12 +233,25 @@ static int list_users(int argc, char *argv[], void *userdata) {
                                                 'b',
                                                 &linger);
                 if (r < 0)
-                        return log_error_errno(r, "Failed to get linger status: %s", bus_error_message(&error, r));
+                        return log_error_errno(r, "Failed to get linger status for user %s: %s",
+                                               user, bus_error_message(&error, r));
+
+                r = sd_bus_get_property_string(bus,
+                                               "org.freedesktop.login1",
+                                               object,
+                                               "org.freedesktop.login1.User",
+                                               "State",
+                                               &error,
+                                               &state);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to get state for user %s: %s",
+                                               user, bus_error_message(&error, r));
```

Comment:
this is race: by the time you query the state here the user might aleady have logged out fully. In which case you'll get SD_BUS_ERROR_UNKNOWN_OBJECT. I'd add a check for that and simply skip the entry in that case.